snapshot.sh

```diff
@@ -4,13 +4,19 @@ set -e
 
 SNAPSHOT_TIME="${SNAPSHOT_TIME:-00:00:00}"
 SNAPSHOT_DAY="${SNAPSHOT_DAY:-*}"
-SNAPSHOT_SIZE="${SNAPSHOT_SIZE:-107374182400}"
 SNAPSHOT_DIR="${SNAPSHOT_DIR:-$PROJECT_ROOT/data}"
 SNAPSHOT_CMD="${SNAPSHOT_CMD:-$@}"
 SNAPSHOT_PATH="${SNAPSHOT_PATH}"
 SNAPSHOT_PREFIX="${SNAPSHOT_PREFIX:-$CHAIN_ID}"
 SNAPSHOT_RETAIN="${SNAPSHOT_RETAIN:-2 days}"
 SNAPSHOT_METADATA="${SNAPSHOT_METADATA:-1}"
+SNAPSHOT_COMPRESS=${SNAPSHOT_COMPRESS:-1}
+SNAPSHOT_FORMAT_DEFAULT=tar
+if [ "$SNAPSHOT_COMPRESS" != "0" ]; then
+  SNAPSHOT_FORMAT_DEFAULT=tar.gz
+fi
+SNAPSHOT_FORMAT="${SNAPSHOT_SAVE_FORMAT:-$SNAPSHOT_FORMAT_DEFAULT}"
```

Comment:
Checking value in acceptable value, if not set to default
https://stackoverflow.com/a/47541882

Can be tested via https://www.onlinegdb.com/online_bash_shell
```bash
SNAPSHOT_SAVE_FORMAT=tar.zstd
valid_snapshot_formats=(tar tar.gz)
if ! printf '%s\0' "${valid_snapshot_formats[@]}" | grep -Fxqz -- $SNAPSHOT_SAVE_FORMAT; then
  SNAPSHOT_SAVE_FORMAT=tar.gz
fi
echo $SNAPSHOT_SAVE_FORMAT
```
```suggestion
SNAPSHOT_SAVE_FORMAT="${SNAPSHOT_SAVE_FORMAT:-$SNAPSHOT_FORMAT}"
valid_snapshot_formats=(tar tar.gz)
# If not one of valid format values, set it to default value
if ! printf '%s\0' "${valid_snapshot_formats[@]}" | grep -Fxqz -- $SNAPSHOT_SAVE_FORMAT; then
  SNAPSHOT_SAVE_FORMAT=tar.gz
fi
```

Reply:
Thank you @PikachuEXE !

`grep -qiw` does it well and is also case-insensitive, so one can drop `.TaR.Gz` :-)

https://github.com/ovrclk/cosmos-omnibus/pull/262/commits/a2ca21eb180791b40ff1962e64d086075109d828#diff-fcde401243107bd79d40cc24b25acc90a4ddfdcfdf38e0d5d5c933d49b41ec9bR16

Reply:
Oh if you want `TaR.Gz` to be accepted, the conditional below should also be updated :S 
Right now it's using `"$SNAPSHOT_SAVE_FORMAT" == "tar.gz"` which is case sensitive

`"${SNAPSHOT_SAVE_FORMAT,,}" == "tar.gz"` is needed (`SNAPSHOT_SAVE_FORMAT` value converted to lower case) [(source)](https://stackoverflow.com/a/19411918/838346)

```bash
SNAPSHOT_SAVE_FORMAT="TaR.Gz"
valid_snapshot_formats=(tar tar.gz)
# If not one of valid format values, set it to default value
if ! echo ${valid_snapshot_formats[@]} | grep -qiw -- "$SNAPSHOT_SAVE_FORMAT"; then
  SNAPSHOT_SAVE_FORMAT=tar.gz
fi

echo $SNAPSHOT_SAVE_FORMAT

# case sensitive
if [ "$SNAPSHOT_SAVE_FORMAT" == "tar.gz" ]; then
    echo GZ
fi

# case sensitive but var value lowercased
if [ "${SNAPSHOT_SAVE_FORMAT,,}" == "tar.gz" ]; then
    echo GZ
fi
```